---
title: "Gene Coexpression"
author: "Joshua H. Cook"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

In this vignette, I demonstrate how `ggasym` can be used to show gene correlation data.

```{r load_pkgs, warning=FALSE, message=FALSE}
library(ggasym)
library(ggplot2)
library(tibble)
library(dplyr)
library(magrittr)
library(purrr)
```


## Introduction

*(Anybody familiar with general molecular biology can skip the Introduction.)*

Genes in an organism's genome are expressed to produce RNA which can have a function of their own and/or be used to create protein. Each cell expresses the thousands of genes at different rates, controlling the function of the cell. Often, biologists are interested in the levels to understand differences between two or more biological groups (such as normal and tumor cells). Further, measuring *correlation*, that is the simillarity in the change of expression of two genes across samples, suggests an interaction between the two genes' functions.

## Data Preparation

I have gene expression (from a microarray) data of the livers of 48 mice, half of which were fed a high-fat diet (making two groups: "High" and "Normal"). I want to find genes that correlated in their change in expression across the samples.

I began by reading in the gene expression data of the livers of mice fed a high fat diet for up to 24 weeks (GEO GDS6248). I prepared the downloaded data before hand to keep this vignette simple, but you can see what I did on this package's GitHub repository at "data-raw/prepare_mouse_gene_expr.R". The major filtration step I completed beforehand was to keep only the top 25% most variable genes to prevent domination of the correlation analysis by genes with little to no change in expression (and to decrease the size of the RData file). 

We can access the data table containing the diets of each mouse in the `mouse_diet` tibble.

```{r load_diet}
mouse_diet
```

And the gene expression data is held in the tibble `moexpr_data`. The first column holds the gene name and all of the other columns are the gene expression for each mouse.

```{r moexpr_data}
moexpr_data
```

## Correlation

I next wanted to find the correlation of each gene compared against the others across mice. For that, I used the `cor.test` function. I initated `cov_tib` with two columns containing each gene-to-gene comparison.

```{r cor_calc}
# calculate the correlation between gene x and gene y
moexpr_correlation <- function(x, y, expr_data) {
    x_data <- filter(expr_data, gene == x) %>% select(-gene) %>% unlist()
    y_data <- filter(expr_data, gene == y) %>% select(-gene) %>% unlist()
    cor.test(x_data, y_data)
}

cov_tib <- as_tibble(expand.grid(moexpr_data$gene, moexpr_data$gene)) %>%
    dplyr::rename(gene_a = "Var1", gene_b = "Var2") %>%
    filter(gene_a != gene_b) %>%
    sample_n(1000) %>%
    mutate(gene_cor = map2(gene_a, gene_b, moexpr_correlation,
                           expr_data = moexpr_data),
           pearson_rho = map_dbl(gene_cor, ~ .x$estimate),
           p_val = map_dbl(gene_cor, ~ .x$p.value))
```

To limit the number of false positives, I used the Benjamini-Hochberg FDR corrected p-value (a.k.a. q-value), then only kept the genes with a value below $0.05$, deeming those as "ssignificantly" correlated. I also only kept the genes with Pearson $\rho$ correlation values above $0.7$ or below $-0.7$ (this is an arbitrary cut-off to narrow down the genes being inspect).

```{r p_corection}
cov_tib <- cov_tib %>%
    mutate(q_val = p.adjust(p_val, method = "BH")) %>%
    filter(q_val < 0.05 & abs(pearson_rho) > 0.7)
```

## Plotting the Results

Finally, I can plot the results in a symmetrix matrix using `geom_asymmat`. I will use the $-log_{10}(\text{q-value})$ and Pearson $\rho$ to fill the top-left and bottom-right triangle, respectively.

```{r plot}
ggplot(cov_tib, aes(x = gene_a, y = gene_b)) +
    geom_asymmat(aes(fill_tl = -log10(q_val), fill_br = pearson_rho)) +
    scale_fill_tl_gradient(low = "lightpink", high = "tomato") +
    scale_fill_br_gradient2(low = "dodgerblue", high = "red")
```

