<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Gene Coexpression • ggasym</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Gene Coexpression">
<meta property="og:description" content="">
<meta property="og:image" content="/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggasym</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9004</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/gene-coexpression.html">Gene Coexpression</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/jhrcook/ggasym">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Gene Coexpression</h1>
                        <h4 class="author">Joshua H. Cook</h4>
            
            <h4 class="date">2019-03-08</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/jhrcook/ggasym/blob/master/vignettes/gene-coexpression.Rmd"><code>vignettes/gene-coexpression.Rmd</code></a></small>
      <div class="hidden name"><code>gene-coexpression.Rmd</code></div>

    </div>

    
    
<p>In this vignette, I demonstrate how <code>ggasym</code> can be used to show gene correlation data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggasym)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(ggplot2)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(tibble)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(dplyr)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(magrittr)
<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(purrr)</code></pre></div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><em>(Anybody familiar with general molecular biology can skip the Introduction.)</em></p>
<p>Genes in an organism’s genome are expressed to produce RNA which can have a function of their own and/or be used to create protein. Each cell expresses the thousands of genes at different rates, controlling the function of the cell. Often, biologists are interested in the levels to understand differences between two or more biological groups (such as normal and tumor cells). Further, measuring <em>correlation</em>, that is the similarity in the change of expression of two genes across samples, suggests an interaction between the two genes’ functions.</p>
</div>
<div id="data-preparation" class="section level2">
<h2 class="hasAnchor">
<a href="#data-preparation" class="anchor"></a>Data Preparation</h2>
<p>I have gene expression (from a microarray) data of the livers of 48 mice, half of which were fed a high-fat diet (making two groups: “High” and “Normal”). I want to find genes that correlated in their change in expression across the samples.</p>
<p>I began by reading in the gene expression data of the livers of mice fed a high fat diet for up to 24 weeks (GEO GDS6248). I prepared the downloaded data before hand to keep this vignette simple, but you can see what I did on this package’s GitHub repository at “data-raw/prepare_mouse_gene_expr.R”. The major filtration step I completed beforehand was to keep only the top 25% most variable genes to prevent domination of the correlation analysis by genes with little to no change in expression (and to decrease the size of the RData file).</p>
<p>We can access the data table containing the diets of each mouse in the <code>mouse_diet</code> tibble.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">mouse_diet
<span class="co">#&gt; # A tibble: 48 x 2</span>
<span class="co">#&gt;    mouse_id  diet  </span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt; </span>
<span class="co">#&gt;  1 GSM994790 Normal</span>
<span class="co">#&gt;  2 GSM994791 Normal</span>
<span class="co">#&gt;  3 GSM994792 Normal</span>
<span class="co">#&gt;  4 GSM994793 Normal</span>
<span class="co">#&gt;  5 GSM994794 Normal</span>
<span class="co">#&gt;  6 GSM994795 Normal</span>
<span class="co">#&gt;  7 GSM994796 Normal</span>
<span class="co">#&gt;  8 GSM994797 Normal</span>
<span class="co">#&gt;  9 GSM994798 Normal</span>
<span class="co">#&gt; 10 GSM994799 Normal</span>
<span class="co">#&gt; # … with 38 more rows</span></code></pre></div>
<p>And the gene expression data is held in the tibble <code>moexpr_data</code>. The first column holds the gene name and all of the other columns are the gene expression for each mouse.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">moexpr_data
<span class="co">#&gt; # A tibble: 4,649 x 49</span>
<span class="co">#&gt;    gene  GSM994790 GSM994791 GSM994792 GSM994793 GSM994794 GSM994795</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt;</span>
<span class="co">#&gt;  1 Aacs       8.82      8.55      9.06     10.5      10.7       9.26</span>
<span class="co">#&gt;  2 Aadac      8.29      8.71      8.48      8.58      8.33      8.35</span>
<span class="co">#&gt;  3 Aadat      8.47      8.33      8.37      8.21      8.21      8.13</span>
<span class="co">#&gt;  4 Aagab      9.93      9.93     10.1       9.96     10.1      10.0 </span>
<span class="co">#&gt;  5 Aamdc      9.57      9.55     10.0       9.89     10.3      10.3 </span>
<span class="co">#&gt;  6 Aard       7.48      7.47      7.77      7.99      8.00      8.05</span>
<span class="co">#&gt;  7 Aars      11.7      11.7      11.8      11.8      11.9      11.8 </span>
<span class="co">#&gt;  8 Aars…      8.86      8.78      8.99      8.67      8.82      8.60</span>
<span class="co">#&gt;  9 Aasdh      9.38      9.49      9.32      9.46      9.51      9.59</span>
<span class="co">#&gt; 10 Aass       9.31      9.81      9.51      8.23      8.33      8.70</span>
<span class="co">#&gt; # … with 4,639 more rows, and 42 more variables: GSM994796 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994797 &lt;dbl&gt;, GSM994798 &lt;dbl&gt;, GSM994799 &lt;dbl&gt;, GSM994800 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994801 &lt;dbl&gt;, GSM994802 &lt;dbl&gt;, GSM994803 &lt;dbl&gt;, GSM994804 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994805 &lt;dbl&gt;, GSM994806 &lt;dbl&gt;, GSM994807 &lt;dbl&gt;, GSM994808 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994809 &lt;dbl&gt;, GSM994810 &lt;dbl&gt;, GSM994811 &lt;dbl&gt;, GSM994812 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994813 &lt;dbl&gt;, GSM994814 &lt;dbl&gt;, GSM994815 &lt;dbl&gt;, GSM994816 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994817 &lt;dbl&gt;, GSM994818 &lt;dbl&gt;, GSM994819 &lt;dbl&gt;, GSM994820 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994821 &lt;dbl&gt;, GSM994822 &lt;dbl&gt;, GSM994823 &lt;dbl&gt;, GSM994824 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994825 &lt;dbl&gt;, GSM994826 &lt;dbl&gt;, GSM994827 &lt;dbl&gt;, GSM994828 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994829 &lt;dbl&gt;, GSM994830 &lt;dbl&gt;, GSM994831 &lt;dbl&gt;, GSM994832 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994833 &lt;dbl&gt;, GSM994834 &lt;dbl&gt;, GSM994835 &lt;dbl&gt;, GSM994836 &lt;dbl&gt;,</span>
<span class="co">#&gt; #   GSM994837 &lt;dbl&gt;</span></code></pre></div>
</div>
<div id="correlation" class="section level2">
<h2 class="hasAnchor">
<a href="#correlation" class="anchor"></a>Correlation</h2>
<p>I next wanted to find the correlation of each gene compared against the others across mice. For that, I used the <code>cor.test</code> function. I initiated <code>cov_tib</code> with two columns containing each gene-to-gene comparison.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># calculate the correlation between gene x and gene y</span>
moexpr_correlation &lt;-<span class="st"> </span><span class="cf">function</span>(x, y, expr_data) {
    x_data &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(expr_data, gene <span class="op">==</span><span class="st"> </span>x) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>gene) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>()
    y_data &lt;-<span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(expr_data, gene <span class="op">==</span><span class="st"> </span>y) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span>(<span class="op">-</span>gene) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unlist">unlist</a></span>()
    <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/cor.test">cor.test</a></span>(x_data, y_data)
}

cov_tib &lt;-<span class="st"> </span><span class="kw"><a href="https://tibble.tidyverse.org/reference/as_tibble.html">as_tibble</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/expand.grid">expand.grid</a></span>(moexpr_data<span class="op">$</span>gene, moexpr_data<span class="op">$</span>gene)) <span class="op">%&gt;%</span>
<span class="st">    </span>dplyr<span class="op">::</span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/select.html">rename</a></span>(<span class="dt">gene_a =</span> <span class="st">"Var1"</span>, <span class="dt">gene_b =</span> <span class="st">"Var2"</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(gene_a <span class="op">!=</span><span class="st"> </span>gene_b) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/sample.html">sample_n</a></span>(<span class="dv">100</span>) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">gene_cor =</span> <span class="kw"><a href="https://purrr.tidyverse.org/reference/map2.html">map2</a></span>(gene_a, gene_b, moexpr_correlation,
                           <span class="dt">expr_data =</span> moexpr_data),
           <span class="dt">pearson_rho =</span> <span class="kw"><a href="https://purrr.tidyverse.org/reference/map.html">map_dbl</a></span>(gene_cor, <span class="op">~</span><span class="st"> </span>.x<span class="op">$</span>estimate),
           <span class="dt">p_val =</span> <span class="kw"><a href="https://purrr.tidyverse.org/reference/map.html">map_dbl</a></span>(gene_cor, <span class="op">~</span><span class="st"> </span>.x<span class="op">$</span>p.value))</code></pre></div>
<p>To limit the number of false positives, I used the Benjamini-Hochberg FDR corrected p-value (a.k.a. q-value), then only kept the genes with a value below <span class="math inline">\(0.05\)</span>, deeming those as “significantly” correlated. I also only kept the genes with Pearson <span class="math inline">\(\rho\)</span> correlation values above <span class="math inline">\(0.7\)</span> or below <span class="math inline">\(-0.7\)</span> (this is an arbitrary cut-off to narrow down the genes being inspect).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov_tib &lt;-<span class="st"> </span>cov_tib <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span>(<span class="dt">q_val =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/stats/topics/p.adjust">p.adjust</a></span>(p_val, <span class="dt">method =</span> <span class="st">"BH"</span>)) <span class="op">%&gt;%</span>
<span class="st">    </span><span class="kw"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(q_val <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/MathFun">abs</a></span>(pearson_rho) <span class="op">&gt;</span><span class="st"> </span><span class="fl">0.7</span>)</code></pre></div>
</div>
<div id="plotting-the-results" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-results" class="anchor"></a>Plotting the Results</h2>
<p>Finally, I can plot the results in a symmetric matrix using <code>geom_asymmat</code>. I will use the <span class="math inline">\(-log_{10}(\text{q-value})\)</span> and Pearson <span class="math inline">\(\rho\)</span> to fill the top-left and bottom-right triangle, respectively.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">asym_cov_tib &lt;-<span class="st"> </span><span class="kw"><a href="../reference/asymmetrise.html">asymmetrise</a></span>(cov_tib, gene_a, gene_b)
<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(asym_cov_tib, <span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">x =</span> gene_a, <span class="dt">y =</span> gene_b)) <span class="op">+</span>
<span class="st">    </span><span class="kw"><a href="../reference/geom_asymmat.html">geom_asymmat</a></span>(<span class="kw"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="dt">fill_tl =</span> <span class="op">-</span><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/Log">log10</a></span>(q_val), <span class="dt">fill_br =</span> pearson_rho)) <span class="op">+</span>
<span class="st">    </span><span class="kw"><a href="../reference/scale_gradient.html">scale_fill_tl_gradient</a></span>(<span class="dt">low =</span> <span class="st">"lightpink"</span>, <span class="dt">high =</span> <span class="st">"tomato"</span>) <span class="op">+</span>
<span class="st">    </span><span class="kw"><a href="../reference/scale_gradient.html">scale_fill_br_gradient2</a></span>(<span class="dt">low =</span> <span class="st">"dodgerblue"</span>, <span class="dt">high =</span> <span class="st">"red"</span>)</code></pre></div>
<p><img src="gene-coexpression_files/figure-html/plot-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#data-preparation">Data Preparation</a></li>
      <li><a href="#correlation">Correlation</a></li>
      <li><a href="#plotting-the-results">Plotting the Results</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Joshua H Cook.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
